### PR TITLE
Fix torch.bucketize docs for "right"

### LINKED ISSUE #91580: Documentation of `bucketize()` is wrong, and is contradicting itself.
### 📚 The doc issue

The documentation of the [bucketize function](https://pytorch.org/docs/stable/generated/torch.bucketize.html), specifically the `right` argument, is wrong. I mean, is it wrong or am I going mad? It says if "right is False (default), then the left boundary is closed". While this makes sense, it's in direct contradiction with the reference table directly below that sentence, and the actual behaviour of the function (which agrees with the reference table but contradicts the aforementioned sentence).

As a further complaint, while that sentence "if right is False (default), then the left boundary is closed" agrees with Numpy's equivalent [digitize function](https://numpy.org/doc/stable/reference/generated/numpy.digitize.html), the actual behaviour is the opposite of Numpy's `digitize()` (with respect to the `right` argument). Why??? Why not just make them behave the same way??????

### Suggest a potential alternative/fix

Ideally I'd like the behaviour `bucketize()` to change (slightly), so that it agrees with the (currently false) sentence "if right is False (default), then the left boundary is closed" in the documentation, and as such agrees with the behaviour of Numpy's `digitize()`, and **not** with the reference table also in the documentation, but I know that's probably not going to happen because, you know, it's behaviour-changing. But let's at least change that sentence in the documentation so that it doesn't contradict itself. That sentence should read "if `right` is `False` (default), then the _right_ boundary is closed".

By the way I apologise if it turns out I'm the one in the wrong. I've stared at that sentence for 2 hours now and it still seems wrong to me.

cc @svekars @carljparker

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #104474

The docs correctly (i.e matching actual op behavior) state that

`right = False` means `boundaries[i-1] < input[m][n]...[l][x] <= boundaries[i]`.

However they previously stated that
`If 'right' is False (default), then the left boundary is closed.`

which contradicts the `boundaries[i-1] < input[m][n]...[l][x] <= boundaries[i]` statement.

This modifies the docs to say `... then the left boundary is OPEN.` and also clarifies that this is the opposite behavior of numpy.digitize.

Fixes #91580

### CODE REVIEW

**Review by aakhundov — APPROVED** [review]
Thanks for the fix!

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -13787,8 +13787,10 @@ def merge_dicts(*dicts):
 
 Returns the indices of the buckets to which each value in the :attr:`input` belongs, where the
 boundaries of the buckets are set by :attr:`boundaries`. Return a new tensor with the same size
-as :attr:`input`. If :attr:`right` is False (default), then the left boundary is closed. More
-formally, the returned index satisfies the following rules:
+as :attr:`input`. If :attr:`right` is False (default), then the left boundary is open. Note that
+this behavior is opposite the behavior of
+`numpy.digitize <https://docs.scipy.org/doc/numpy/reference/generated/numpy.digitize.html>`_
```
Comment (svekars) [inline torch/_torch_docs.py:13792]:
```suggestion
`numpy.digitize <https://docs.scipy.org/doc/numpy/reference/generated/numpy.digitize.html>`_.
```

**Review by svekars — APPROVED** [review]
One nit, but LGTM! Check the output [here](https://docs-preview.pytorch.org/pytorch/pytorch/104474/generated/torch.bucketize.html#torch-bucketize) as well.

**Pull request closed without merging**